pgxn/neon/pagestore_smgr.c

```diff
@@ -2769,11 +2850,16 @@ neon_redo_read_buffer_filter(XLogReaderState *record, uint8 block_id)
 	LWLock	   *partitionLock;
 	Buffer		buffer;
 	bool		no_redo_needed;
-	BlockNumber relsize;
 
 	if (old_redo_read_buffer_filter && old_redo_read_buffer_filter(record, block_id))
 		return true;
 
+	/*
+	 * Always apply record with full page image
+	 */
+	if (XLogRecBlockImageApply(record, block_id))
+		return false;
+
```

Comment:
This change is actually not required to fix the problem.
But I still think that it will be wasteful  just to throw away received full page image.
If customer has created read-only replica, then most likely he is going to use it.
Either for load balancing of read-only queries, either for fast replacing of crashed master (which we do not support now, right?)
In any case - replica needs to be synced with master as much as possible.
Probability that pages updated at master are not needed at replica is quite small.

If shared buffers are small and can not fit all working set, then we have LFC which can hold received pages.

So I think that should safe network traffic (because otherwise page will be transferred twice) and avoid extra `get_page` requests to page server.

Reply:
But as this optimisation has no relation to the problem, I removed this patch from this PR